github/client.go

```diff
@@ -152,6 +152,67 @@ func (c *Client) ReadReleaseByTag(ctx context.Context, tkn string, repoURL url.U
 	return model.ToSvcGithubRelease(rls)
 }
 
+// TODO once svcmodel.Release has git tag field, no gitTag parameter will be needed
+func (c *Client) UpdateRelease(ctx context.Context, tkn string, repo svcmodel.GithubRepo, rls svcmodel.Release, gitTag string) error {
+	// Release can be only updated by release ID
+	// Therefore release ID is fetched by tag name first
+	id, err := c.getReleaseID(ctx, tkn, repo, gitTag)
+	if err != nil {
+		return fmt.Errorf("failed to get release ID: %w", err)
+	}
+
+	_, _, err = c.getGithubClient(tkn).Repositories.EditRelease(
+		ctx,
+		repo.OwnerSlug,
+		repo.RepoSlug,
+		id,
+		&github.RepositoryRelease{
+			Name: &rls.ReleaseTitle,
+			Body: &rls.ReleaseNotes,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to update release: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Client) DeleteRelease(ctx context.Context, tkn string, repo svcmodel.GithubRepo, tagName string) error {
+	// Release can be deleted only by release ID
+	// Therefore release ID is fetched by tag name first
+	id, err := c.getReleaseID(ctx, tkn, repo, tagName)
+	if err != nil {
+		return fmt.Errorf("failed to get release ID: %w", err)
+	}
+
+	_, err = c.getGithubClient(tkn).Repositories.DeleteRelease(ctx, repo.OwnerSlug, repo.RepoSlug, id)
+	if err != nil {
+		return fmt.Errorf("failed to delete release: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Client) getReleaseID(ctx context.Context, tkn string, repo svcmodel.GithubRepo, tagName string) (int64, error) {
+	rls, _, err := c.getGithubClient(tkn).Repositories.GetReleaseByTag(
+		ctx,
+		repo.OwnerSlug,
+		repo.RepoSlug,
+		tagName,
+	)
+	if err != nil {
+		var githubErr *github.ErrorResponse
+		if errors.As(err, &githubErr) && githubErr.Response.StatusCode == http.StatusNotFound {
+			return 0, svcerrors.NewGithubReleaseNotFoundError().Wrap(err)
```

Comment:
If a release is not found, I will return `svcerrors.NewGithubReleaseNotFoundError()`. However, the service will not return this error to the user because, as we agreed, creating, updating, or deleting a GitHub release will be done implicitly. 

The user will simply click the "Populate to GitHub" button. For deleting a release, the user can check a checkbox labeled "Delete GitHub release if it exists" or something similar.